### DISCUSSION TITLE
Setting so user can only see the other users they have added in a user group, not all of them?

### DISCUSSION THREAD

**Discussion by kevadamson** [craftcms/cms#8851] [Ideas]
So a bit like a user can only manage entries they have authored (via the permissions), they can only view, create and manage users they have added to a certain usergroup. Is this possible? It will also mean they can only select users that they have added via the users field type as well. If this isn't possible, does anyone know of a plugin that enables it?

- Craft version: 3.4.25

**Comment by kevadamson** [comment]
Here is an example use case:

A teacher uses a frontend registration form to create an account. There is a user group called "Pupils" which they can add users to, but they can only see the ones they have added.

Make sense?

**Comment by kevadamson** [comment]
@brandonkelly OK, sorry to chime on about this, but I have another quick question: If I was to get someone to create a plugin to do this, would that actually be possible? i.e. Are the hooks/events there for a plugin to be possible, or is this something I will need to wait for as a native system update by the Craft developers themselves? Thanks in advance :)

**Comment by brandonkelly** [comment]
It should be possible by A) keeping track of those user relations somehow, and B) modifying user queries via the `EVENT_AFTER_PREPARE` event on `craft\elements\db\UserQuery`, yes.

**Comment by kevadamson** [comment]
@brandonkelly You're a gent. Many thanks. I was thinking that their must be a record of which user created which user? Am I right in that assumption?

**Comment by brandonkelly** [comment]
There is not, you’d have to track that.

**Comment by kevadamson** [comment]
@brandonkelly But to clarify, there would be a way to do that via A and/or B as suggested above?

**Comment by brandonkelly** [comment]
You need both. Tracking it was step A.
